packages/react-reconciler/src/ReactFiberLazyComponent.js

```diff
@@ -73,7 +73,10 @@ export function readLazyComponentType<T>(lazyComponent: LazyComponent<T>): T {
           }
         },
       );
-      lazyComponent._result = thenable;
+      // Don't race with a sync thenable
+      if (lazyComponent._status === Pending) {
+        lazyComponent._result = thenable;
```

Comment:
Alt suggestion:

```js
// Check if it resolved synchronously
if (lazyComponent._status === Resolved) {
  return lazyComponent._result;
}
```

Since if it's already resolved we don't need to throw the thenable

Reply:
Aaaah. That's why it had the loading state. Thanks.